examples/root_macros/generate_4body_decay_chain.C

```diff
@@ -0,0 +1,273 @@
+/*----------------------------------------------------------------------------
+ *
+ *   Copyright (C) 2016 - 2017 Antonio Augusto Alves Junior
+ *
+ *   This file is part of Hydra Data Analysis Framework.
+ *
+ *   Hydra is free software: you can redistribute it and/or modify
+ *   it under the terms of the GNU General Public License as published by
+ *   the Free Software Foundation, either version 3 of the License, or
+ *   (at your option) any later version.
+ *
+ *   Hydra is distributed in the hope that it will be useful,
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *   GNU General Public License for more details.
+ *
+ *   You should have received a copy of the GNU General Public License
+ *   along with Hydra.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ *---------------------------------------------------------------------------*/
+
+/*
+ * generate_decay_chain.C
+ *
+ *  Created on: 25/03/2018
+ *      Author: Antonio Augusto Alves Junior
+ */
+
+
+/*---------------------------------
+ * std
+ * ---------------------------------
+ */
+#include <iostream>
+#include <assert.h>
+#include <time.h>
+#include <vector>
+#include <array>
+#include <chrono>
+
+
+/*---------------------------------
+ * Include hydra classes and
+ * algorithms for
+ *--------------------------------
+ */
+#ifndef HYDRA_HOST_SYSTEM
+#define HYDRA_HOST_SYSTEM CPP
+#endif
+
+#ifndef HYDRA_DEVICE_SYSTEM
+#define HYDRA_DEVICE_SYSTEM TBB
+#endif
+
+
+#include <hydra/PhaseSpace.h>
+#include <hydra/Chains.h>
+#include <hydra/Evaluate.h>
+#include <hydra/Function.h>
+#include <hydra/FunctorArithmetic.h>
+#include <hydra/FunctionWrapper.h>
+#include <hydra/Copy.h>
+#include <hydra/Tuple.h>
+#include <hydra/Vector3R.h>
+#include <hydra/Vector4R.h>
+#include <hydra/host/System.h>
+#include <hydra/device/System.h>
+#include <hydra/Placeholders.h>
+
+#include <hydra/functions/PlanesDeltaAngle.h>
+/*-------------------------------------
+ * Include classes from ROOT to fill
+ * and draw histograms and plots.
+ *-------------------------------------
+ */
+#include <TROOT.h>
+#include <TH1D.h>
+#include <TF1.h>
+#include <TH2D.h>
+#include <TH3D.h>
+#include <TApplication.h>
+#include <TCanvas.h>
+#include <TColor.h>
+#include <TString.h>
+#include <TStyle.h>
+
+
+
+using namespace hydra::placeholders;
+
```

Comment:
Please add a concise description about what the example really does... also some comments along the code.